src/Shell/Process.php

```diff
@@ -249,20 +254,94 @@ public function onSuccess(Closure $closure = null)
         return $this;
     }
 
+    /**
+     * @return mixed
+     */
+    public function getStdin()
+    {
+        return $this->pipes[static::STDIN];
+    }
+
+    /**
+     * @param $stdin
+     *
+     * @return $this
+     */
+    public function setStdin($stdin)
```

Comment:
I know there's no CS checks running here, but the param argument needs the type specified (mixed should be avoided). This applies to each of the `setStd[out|in|err]` methods.

At minimum the `setStd[out|in|err]` api needs to require the argument be an array. Best case would be to enumerate the parts so that the user doesn't have to know explicitly what proc_open requires as the format of the descriptorspec.

Reply:
[fixed](https://github.com/nickel715/shell-command/commit/a417c0c8a50a616efa531862c0f745b834233c02#diff-358131510f5b9763f089b505b10e0936R258)